app-cpp/app355.cpp

```diff
@@ -480,7 +369,388 @@ void SatOrbit::OnPrintTrains(const std::vector<std::vector<OrbitalData>>& trainV
+    for (std::size_t i = 0; i < trainVector.size(); ++i)
+    {
+        for (std::size_t j = i + 1; j < trainVector.size(); ++j)
+        {
+            double deltaMotion = abs(trainVector[i][0].GetTLE().GetMeanMotion() - trainVector[j][0].GetTLE().GetMeanMotion());
+            if (deltaMotion < 0.001)
+            {
+                trainVector[i].insert(trainVector[i].end(), trainVector[j].begin(), trainVector[j].end());
+                trainVector.erase(trainVector.begin() + j);
+                --j;
+            }
+        }
+    }
+    
+    return trainVector;
+}
+
+void SatOrbitSingle::OnPrintTrains(const std::vector<std::vector<OrbitalData>>& trainVector)
+{
+    int trainCount = 0;
+    // Print the contents of the train list
+
+    std::for_each(trainVector.begin(), trainVector.end(), [&](auto& train)
+    {
+        std::cout << "   TRAIN #" << trainCount << std::endl;
+        std::cout << "   COUNT: " << train.size() << std::endl;
+
+        std::for_each(train.begin(), train.end(), [](auto& data)
+        {
+            // Print the name, mean motion, latitude, longitude, and altitude
+            std::cout << data.GetTLE().GetName() << ": " << data.GetTLE().GetMeanMotion() << std::endl;
+            std::cout << "Lat: " << data.GetLatitude() << std::endl;
+            std::cout << "Lon: " << data.GetLongitude() << std::endl;
+            std::cout << "Alt: " << data.GetAltitude() << std::endl << std::endl;
+        });
+        std::cout << std::endl << std::endl;
+        trainCount++;
+    });
+}
+
+// Helper
+/*static*/ bool SatOrbitSingle::SortPredicate(const OrbitalData& inLHS, const OrbitalData& inRHS)
+{
+    return inLHS.GetTLE().GetMeanMotion() < inRHS.GetTLE().GetMeanMotion();
+}
+
+#pragma endregion {}
+
+//----------------------------------------
+#pragma region SatOrbitMulti
+
+class SatOrbitMulti : public SatOrbitSingle
+{
+// Interface
+public:
+    SatOrbitMulti(std::size_t inNumThreads) :
+        mNumThreads{inNumThreads}
+    {
+        // Do nothing
+    }
+    ~SatOrbitMulti() override = default;
+    static std::unique_ptr<SatOrbitMulti> Make(std::size_t inThreads = 4);
+
+// Types
+private:
+    using tle_const_iterator = std::vector<sat355::TLE>::const_iterator;
+    using IteratorPairVector = std::vector<std::tuple<orbit_iterator, orbit_iterator>>;
+
+// Implementation
+private:
+    // SatOrbit
+    void OnCalculateOrbitalData(const std::vector<sat355::TLE>& inTLEVector, OrbitalDataVector& ioDataVector) override;
+    void OnSortOrbitalVector(std::vector<OrbitalData>& ioOrbitalVector) override;
+
+    // SatOrbitMulti
+    virtual void OnCalculateOrbitalDataMulti(const tle_const_iterator& tleBegin, const tle_const_iterator& tleEnd, OrbitalDataVector& ioDataVector);
+    virtual void OnSortOrbitalVectorMulti(orbit_iterator& inBegin, orbit_iterator& inEnd);
+    virtual void OnSortMergeVector(orbit_iterator& ioBegin, orbit_iterator& ioMid, orbit_iterator& ioEnd);
```

Comment:
Subjective:
Rename `OnSortMergeVector` to `OnSortMergeVectorMulti` to provide more of a hint that this method is logically grouped with `SatOrbitMulti` functionality.